pkg/apis/pipeline/v1beta1/taskrun_validation.go

```diff
@@ -120,17 +120,49 @@ func (ts *TaskRunSpec) validateInlineParameters(ctx context.Context) (errs *apis
 			Name:    p.Name,
 			Default: &p.Value,
 		}
+		if p.Value.ObjectVal != nil {
+			pSpec.Properties = make(map[string]PropertySpec)
+			prop := make(map[string]PropertySpec)
+			for k := range p.Value.ObjectVal {
+				prop[k] = PropertySpec{Type: p.Value.Type}
+			}
+			pSpec.Properties = prop
+			pSpec.Type = ParamTypeObject
+		}
 		paramSpec = append(paramSpec, pSpec)
 	}
 	for _, p := range ts.TaskSpec.Params {
 		skip := false
-		for _, ps := range paramSpec {
+		for i, ps := range paramSpec {
 			if ps.Name == p.Name {
+				if p.Default != nil && p.Default.ObjectVal != nil {
+					for k, v := range p.Default.ObjectVal {
+						if paramSpec[i].Default.ObjectVal == nil {
+							paramSpec[i].Default.ObjectVal = map[string]string{k: v}
+						} else {
+							paramSpec[i].Default.ObjectVal[k] = v
+						}
+						if paramSpec[i].Properties == nil {
+							paramSpec[i].Properties = map[string]PropertySpec{
+								k: {Type: ParamTypeString},
+							}
+						} else {
+							paramSpec[i].Properties[k] = PropertySpec{Type: ParamTypeString}
+						}
+					}
+				}
```

Comment:
I am wondering if we can change `paramSpec` from `[]ParamSpec` to `map[string]ParamSpec` type (paramname:paramSpec) to avoid the 3-layer for loops. When all ParamSpecs are needed at the end for validation, we can just use a single for loop to get all the values of the map. wdyt?

nit: maybe also rename `paramSpec` to `paramSpecsForValidation` or just add a comment to explain what is this variable for?

Reply:
I can change it to a map, however, it will require changing many other calls too since `[]ParamSpec` is required by `ValidatePipelineParams` and many other functions. They will all need to be changed. Might be outside the scope of just this PR.

Reply:
I can certainly change `paramSpec` to `paramSpecForValidation`.

Reply:
In this function, we just called `ValidateParameterVariables` at the end to delegate the validation job. Since it expects a slice of paramspec for the third argument, we can just collect the map's values using a single loop right before calling this func. This may help us avoid layered loops?

Reply:
I see what you mean. So basically just convert the map to a slice of param spec at the end before passing it for validation.

Reply:
Yes!

Reply:
Done!